### PR TITLE
Add macros for RISCV hwcaps.

### PR DESCRIPTION
**Pull request by michael-roe**

### CODE REVIEW

File: include/internal/hwcaps.h [hunk 1]
```diff
@@ -176,6 +176,14 @@ CPU_FEATURES_START_CPP_NAMESPACE
 #define PPC_FEATURE2_HTM_NO_SUSPEND 0x00080000
 #endif
 
+#define RISCV_HWCAP_A (1UL << ('A' - 'A'))
```
Comment (toor1245) [inline include/internal/hwcaps.h:179]: could you provider link to linux source for consistency?https://elixir.bootlin.com/linux/latest/source/arch/riscv/include/uapi/asm/hwcap.h

Comment (toor1245) [reply include/internal/hwcaps.h:179]: what about `RISCV_HWCAP_I`, `RISCV_HWCAP_A`?

Comment (michael-roe) [reply include/internal/hwcaps.h:179]: Yes, I'll provide a link (I was intending to do that, but forgot) and include i and e.

Comment (gchatelet) [reply include/internal/hwcaps.h:179]: @michael-roe can you add the link please?

Comment (michael-roe) [reply include/internal/hwcaps.h:180]: Ok, done.

**Pull request merged**